ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -230,7 +245,7 @@ private BackupReplayFile newReplayFile() {
     }
 
     return (latestBakFile != null)
-        ? Optional.of(new BackupReplayFile(latestBakFile))
+        ? Optional.of(BackupReplayFile.openFile(latestBakFile))
```

Comment:
I think things get a little complicated here - if the replay file is `NO_VERSION`, then we need to create a new one with the newer version but restore using the existing one. I think it might just be simpler if we don't compare across versions - we didn't have this functionality until 0.13, it's probably OK to go one more deploy without it.

thoughts @stevenpyzhang @rodesai?

Reply:
why do we have multiple backup files?

Reply:
Before 0.13, if the file was ever corrupted we would simply start a new one and keep the old one. This behavior is changing to fail hard if it's corrupted. After this change, if the old file was created before 0.13 we want to write it in a format that doesn't have the problems described in this PR, but we want to leave the old one in place just in case (at least that was my thoughts).

Does that answer your question?

Reply:
I'm fine with dropping the older backup files for this release. We can do this by configuring a new location in cc-spec-ksql. We would be kicking the problem down the road, so at least file an issue that we need to add something to rewrite to the current version if the backup is old.

Reply:
> so at least file an issue that we need to add something to rewrite to the current version if the backup is old.
I don't think we need to do this, the backup file would be written from the beginning of the command topic so we could just ignore the old ones altogether.

Reply:
We still need something to detect that the current version is old and then write out the new file right? (even if we don't delete the file)

Reply:
We have two options: (1) we configure a new location in cc-spec and then the old files are as good as having never existed or (2) we use the header approach, which the very first time it starts up on 0.13 it'll write a new file, and from then on it ignores the old one. Perhaps I don't understand what you meant - what we're kicking down the road?

Reply:
This patch adds versioning to the backup. That means we need to be able to deal with the older versions. I think what you're proposing is that if the current backup file's version is old, we simply throw it away? I'm ok with this to keep things simple. These bumps should be very infrequent.

Reply:
@almog Are we still doing the versioning now we'll ignore the old backups?

Reply:
@spena - feel free to make the call on that. The benefit of adding a header is that in the future if we ever make changes to the functionality we'll have a non-disruptive way forward. If we don't think we'll be making those changes, we can just make it simple.